crypto-core/src/main/java/com/palantir/crypto/cipher/CipherStreamSupplier.java

```diff
@@ -30,12 +35,14 @@
     /**
      * Wraps the {@link InputStream} in a {@link CipherInputStream} which uses the given {@link Cipher} for decryption.
      */
-    CipherInputStream getInputStream(InputStream is, Cipher cipher);
+    CryptoInputStream getInputStream(String transformation, Properties props, InputStream is,
```

Comment:
The `transformation` is the algorithm, e.g., `"AES/CBC/PKCS5Padding"`

One idea is to make it take an `InputStream` and a `SeekableCipher`, and then get the `Key` and `AlgorithmParameterSpec` that we need from that. Could also get the algorithm from the `SeekableCipher` if we were to add a `getAlgorithm()` method to it, which seems reasonable enough to me. Would that be better?